### PR TITLE
compile: Allow object generating rules to be annotated as entrypoint

### PR DESCRIPTION
**Pull request by johanfylling**
Using ground ref of rule when generating entrypoint from annotation as variable in head suffix cannot be included in generated entrypoint path.

Fixes: #5577
Signed-off-by: Johan Fylling <johan.dev@fylling.se>

### CODE REVIEW

File: compile/compile.go [hunk 1]
```diff
@@ -234,7 +234,7 @@ func addEntrypointsFromAnnotations(c *Compiler, ar []*ast.AnnotationsRef) error
 				}
 			case "rule":
 				if r := ref.GetRule(); r != nil {
-					entrypoint = r.Ref()
+					entrypoint = r.Path()
```
Comment (johanfylling) [inline compile/compile.go:237]: `Rule.Path()` is deprecated; but here we have a clear use for a rule path that only includes ground.

So the question is, should `Path()` be un-deprecated with an updated godoc; or should we perhaps duplicate the behavior under a different name to make mistakes less likely?

Comment (ashutosh-narkar) [reply compile/compile.go:237]: Can't we use `r.Ref().GroundPrefix()` ?

Comment (ashutosh-narkar) [reply compile/compile.go:237]: Or you can directly use the `Extend` call?

Comment (philipaconrad) [reply compile/compile.go:237]: @ashutosh-narkar That's actually what `Rule.Path` does, I think ([ast/policy source](https://github.com/open-policy-agent/opa/blob/4bc74912372fd5479cb8700074582723316d3b06/ast/policy.go#L600-L608)): it wraps an `Extend` call around the ground prefix of the ref.

Comment (srenatus) [reply compile/compile.go:237]: Re: GroundPrefix() -- as long as we check edge cases like this
```rego
p.foo.bar if true
p.foo[x] if x := 12
```
we should be OK there 🤔

**Review by ashutosh-narkar — APPROVED** [review]

**Pull request merged**